### PR TITLE
Added control for checking secretes in env variables of Agents in Agent pool

### PR DESCRIPTION
**Pull request by v-himkam**

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 1]
```diff
@@ -215,4 +215,79 @@ class AgentPool: ADOSVTBase
...
         return $controlResult
     }
+
+    hidden [ControlResult] CheckCredInEnvironmentVariables([ControlResult] $controlResult)
+    {
+        try 
+        {   
+            $agentPoolsURL = "https://dev.azure.com/{0}/{1}/_settings/agentqueues?queueId={2}&__rt=fps&__ver=2" -f $($this.SubscriptionContext.SubscriptionName), $this.ProjectId ,$this.AgentPoolId;
+            $agentPool = [WebRequestHelper]::InvokeGetWebRequest($agentPoolsURL);
+            $patterns = $this.ControlSettings.Patterns | where {$_.RegexCode -eq "SecretsInBuild"} | Select-Object -Property RegexList;
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:225]: Keep this logic only if $agentPool response object has some data

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 2]
```diff
@@ -215,4 +215,79 @@ class AgentPool: ADOSVTBase
...
+                                        break
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    if($noOfCredFound -eq 0) 
+                    {
+                        $controlResult.AddMessage([VerificationResult]::Passed, "No secrets found in environment variables of Agents in Agent pool.");
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:256]: lower case

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 1 above)
Comment (v-himkam) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1]: As discussed with no need for this check

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 2 above)
Comment (v-himkam) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1]: done

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**